src/wp-includes/class-wp-query.php

```diff
@@ -3066,17 +3073,38 @@ public function get_posts() {
 		";
 
 		$this->request = $old_request;
+		$this->count_request = "
+			SELECT COUNT(DISTINCT {$count_field})
+			FROM {$wpdb->posts} $join
+			WHERE 1=1 $where
+		";
 
 		if ( ! $q['suppress_filters'] ) {
 			/**
 			 * Filters the completed SQL query before sending.
 			 *
 			 * @since 2.0.0
+			 * @since x.x.x This query no longer contains a `SQL_CALC_FOUND_ROWS` modifier by default.
 			 *
 			 * @param string   $request The complete SQL query.
 			 * @param WP_Query $query   The WP_Query instance (passed by reference).
 			 */
 			$this->request = apply_filters_ref_array( 'posts_request', array( $this->request, &$this ) );
+
+			if ( false !== strpos( $this->request, 'SQL_CALC_FOUND_ROWS' ) ) {
```

Comment:
```suggestion
			if ( str_contains( $this->request, 'SQL_CALC_FOUND_ROWS' ) ) {
```

Polyfilled in WP

https://github.com/WordPress/wordpress-develop/blob/01623b164cd7c367135422d212414f8bd16338a9/src/wp-includes/compat.php#L427-L443

Or possibly `str_starts_with()` -- also polyfilled.